src/components/Card.js

```diff
@@ -1,30 +1,59 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleLikeDelete
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleLikeDelete = handleLikeDelete;
   }
 
   _setEventListeners() {
     this._cardImageElement.addEventListener("click", () =>
       this._handleImageClick({ name: this._name, link: this._link })
     );
-    this._likeButton.addEventListener("click", () => this._handleLikeButton());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteButton()
-    );
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeButton();
+      if (!this._likeButton.classList.contains("card__like-button_active")) {
+        this._handleLikeDelete(this._id);
+      } else if (
+        this._likeButton.classList.contains("card__like-button_active")
+      ) {
+        this._handleLikeClick(this._id);
+      }
+    });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
   }
 
-  _handleDeleteButton() {
+  remove() {
     this._cardElement.remove();
-    this._cardElement = null;
+    // this._cardElement = null;
   }
 
   _handleLikeButton() {
     this._likeButton.classList.toggle("card__like-button_active");
   }
 
+  _getCardLike() {
```

Comment:
Needs correction: please rename this method to `renderLikes` as it's not a getter